### ISSUE TITLE
TypeError: Cannot read properties of undefined (reading '$$') when using 0.42.0 or newer

### ISSUE THREAD

**Issue by Trystan-SA** [carbon-design-system/carbon-components-svelte#896]
When I'm using any components from carbon-components-svelte. It trigger the following error :

![chrome_bxTO9mFTBa](https://user-images.githubusercontent.com/43876044/140764181-445a2468-e8e8-4296-a3cc-2d18c5bc97c1.png)
![chrome_YRSjjmqjap](https://user-images.githubusercontent.com/43876044/140764192-fccb649e-3ee3-4af7-a3c3-55e50daff6c7.png)

Here is the code that trigger the error. It happen when I'm using any components (buttons, forms, accordions): 
![Code_R6CQCZ0gOO](https://user-images.githubusercontent.com/43876044/140764284-dc4f96c5-05a4-441c-b0fb-10e513e9d5ec.png)

I don't have this issue if I'm using the 0.41.0 version. It start breaking at 0.42.0 or newer.

My package versions :
svelte: 3.44.1
esbuild: 0.13.12
esbuild-svelte: 0.5.7  (and tested with 0.5.3 too)

**Comment by ulvido** [comment]
same environment and same error for me too. 
in my case, I send a component with a javascripts general [Custom Event ](https://developer.mozilla.org/en-US/docs/Web/Events/Creating_and_triggering_events#adding_custom_data_%E2%80%93_customevent) and catch the event at another place. 

The error occurs if you want to render a received component with `svelte:component`

#### EXAMPLE CODE

##### create a custom event and dispatch at some far place
```javascript
...

const register = new CustomEvent(
  "REGISTER_APP",
  {
    bubbles: true,
    detail: {
      mycomponent: SomeSvelteComponent,
    },
  }
)

window.dispatchEvent(register);
```

##### catch the custom event at store
```javascript
...

export const customComponent = writable(undefined)

window.addEventListener("REGISTER_APP", e => {
  customComponent.set(e.detail.mycomponent)
});

...
```

##### use component
```javascript
...

import { customComponent } from "./store"

<div>
  {#if $customComponent}
     <svelte:component this={$customComponent} />
  {/if}
</div>

...
```
![image](https://user-images.githubusercontent.com/27722274/142061097-ce24355a-b8c0-4392-9c80-1f9e6a3937c8.png)

something fails when setting the component but what that I dont know yet.
![image](https://user-images.githubusercontent.com/27722274/142061711-92ca5bef-dbec-43d5-9ddb-7d3043404ce5.png)

**Comment by onelesd** [comment]
Can confirm the same error with v0.42.0 or higher. My env:

svelte: 3.44.1
esbuild: 0.14.8
esbuild-svelte: 0.6.0

**Comment by metonym** [comment]
I recall experiencing a similar error when using a Svelte library that ships bundled ESM code in addition to uncompiled Svelte source files.

Just a hunch, but I wonder if it is trying to use the bundled Carbon Svelte library instead of the uncompiled source.

**Comment by onelesd** [comment]
Related issue: https://github.com/sveltejs/svelte/issues/6584

**Comment by metonym** [comment]
I was able to repro the error using esbuild + esbuild-svelte + carbon-components-svelte.

I definitely think it's related to `esbuild` using the compiled source code. When I manually changed the "main" field to point to the uncompiled Svelte source code, it worked.

```diff
- "main": "./lib/index.js",
- "module": "./lib/index.mjs",
+ "main": "./src/index.js",
```

It should be prudent to drop support for the bundled source code, and instead point the "svelte" and "main" fields to the Svelte source code. This practice also seems to be encouraged by `svelte-kit package`.
